packages/x-scheduler/src/joy/event-calendar/EventCalendar.types.ts

```diff
@@ -11,12 +21,69 @@ export interface EventCalendarProps extends React.HTMLAttributes<HTMLDivElement>
    * Callback fired when some event of the calendar change.
    */
   onEventsChange?: (value: CalendarEvent[]) => void;
-  /**
-   * Translation overrides for the component's texts.
-   */
-  translations?: Partial<SchedulerTranslations>;
   /**
    * The resources that can be assigned to events.
    */
   resources?: CalendarResource[];
+  /**
+   * The view currently displayed in the calendar.
+   */
+  view?: EventCalendarView;
+  /**
+   * The view initially displayed in the calendar.
+   * To render a controlled calendar, use the `view` prop.
+   * @default "week"
+   */
+  defaultView?: EventCalendarView;
+  /**
+   * Event handler called when the view changes.
+   */
+  onViewChange?: (view: EventCalendarView, event: React.UIEvent | Event) => void;
+  /**
+   * The date currently displayed in the calendar.
+   */
+  visibleDate?: SchedulerValidDate;
+  /**
+   * The date initially displayed in the calendar.
+   * To render a controlled calendar, use the `visibleDate` prop.
+   * @default today
+   */
+  defaultVisibleDate?: SchedulerValidDate;
+  /**
+   * Event handler called when the visible date changes.
+   */
+  onVisibleDateChange?: (visibleDate: SchedulerValidDate, event: React.UIEvent) => void;
 }
+
+export interface EventCalendarInstance {
+  /**
+   * Set the view of the calendar.
+   */
+  setView: (view: EventCalendarView, event: React.UIEvent | Event) => void;
+  /**
+   * Update an event in the calendar.
+   */
+  updateEvent: (calendarEvent: CalendarEvent) => void;
+  /**
+   * Delete an event from the calendar.
+   */
+  deleteEvent: (eventId: CalendarEventId) => void;
+  /**
+   * Go to today's date without changing the view.
+   */
+  setVisibleDateToToday: (event: React.UIEvent) => void;
```

Comment:
The name of `setVisibleDateToToday` / `goToPreviousVisibleDate` / `goToNextVisibleDate` and `goToDay` are very imperfect.
If you have better proposals I'm interested.

Reply:
Updated :ok_hand:

Reply:
We could do like Full Calendar which allows to change the visible date inside the method that changes the view ([see the doc](https://fullcalendar.io/docs/Calendar-changeView))

```ts
calendar.changeView('timeGridDay', '2017-06-01');
```